mirascope/rag/chunkers.py

```diff
@@ -0,0 +1,31 @@
+"""Chunkers for the RAG module."""
+import uuid
+from typing import Callable, Concatenate
+
+from pydantic import BaseModel
+
+from .types import Document
+
+
+def base_chunk_fn(
+    text: str,
+    chunk_size: int = 1000,
+    chunk_overlap: int = 0,
+) -> list[Document]:
+    """Chunk the text into chunks of size `chunk_size`."""
+    chunks: list[Document] = []
+    start: int = 0
+    while start < len(text):
+        end: int = min(start + chunk_size, len(text))
+        chunks.append(Document(text=text[start:end], id=str(uuid.uuid4())))
+        start += chunk_size - chunk_overlap
+    return chunks
+
+
+class BaseChunker(BaseModel):
+    """"""
+
+    chunk_fn: Callable[Concatenate[str, ...], list[Document]] = base_chunk_fn
+
+    def chunk(self, text: str) -> list[Document]:
```

Comment:
I'm thinking we should make this an abstract method and drop `chunk_fn` entirely i.e. require a subclass implement the `chunk` function.

If we want, we can include something like `SimpleChunker` or `DumbChunker` that actually implements `base_chunk_fn` as an OOB class. Thoughts?

Reply:
`rag/chunkers/text_chunker` ?

Reply:
Do we want to then import `from mirascope.rag.chunkers import TextChunker` or do we just want to example like you current have for the wikipedia store?

The former would mean importing the chunker from mirascope instead of defining it in the example

Reply:
they can import premade from chunks e.g. textchunker or import basechunker for extending purposes. the textchunker can then be recursive chunker or whatever chunker

Reply:
Ok, then update the example to import `TextChunker` once refactored